docs/source/architecture/events_and_transactions_receipts.rst

```diff
@@ -0,0 +1,343 @@
+*******************************
+Events and Transaction Receipts
+*******************************
+
+Hyperledger Sawtooth supports the creating and broadcasting of Events.
+This allows application developers to do the following:
+
+- Subscribe to events that occur related to the blockchain, such
+  as a new block being committed or switching to a new fork.
+- Subscribe to application specific events defined by a transaction family.
+- Relay information about the execution of a transaction back to
+  clients without storing that data in state.
+
+It is important to note the following constraints:
+
+- Events are transmitted only on block boundaries.
+- Event notifications are implemented internally using the Observer pattern.
+- Both internal and external event subscribers are supported.
```

Comment:
Delete the constraints section, I don't think the last two bullets are accurate anymore.